src/TTGamesExplorerRebirthUI/Forms/MainForm.cs

```diff
@@ -11,9 +11,24 @@ public partial class MainForm : DarkForm
         private string _loadedGameFolderPath = "";
         private GamesMetadata _gameMetadata;
 
-        public MainForm()
+        public MainForm(string path = null)
         {
             InitializeComponent();
+
+            if (path != null)
+            {
+                if (Directory.Exists(path))
+                {
+                    AppSettings.Instance.GameFolderPath = path;
+                    return;
+                }
+
+                new Thread(() =>
+                {
+                    Thread.CurrentThread.IsBackground = true;
+                    Helper.OpenFileInternal(path, Path.GetFullPath(path));
+                }).Start();
```

Comment:
Are you sure the Thread is necessary ? What about just calling `Helper.OpenFileInternal(path, Path.GetFullPath(path));` instead ?

Reply:
That would block the mainform from opening. The idea is that it will open the file in a new window and then open the immediate directory of the file in the MainForm, but that is not implemented yet.